### PR TITLE
Add .oct-config files to specify source is utf-8

### LINKED ISSUE #251: error: regexp: unrecognized character after (? or (?- at position 6 of expression
I got this doctesting Symbolic on Fedora 36.  It did not happen when I build on upcoming Fedora 37.

This is Fedora's Octave 6.4.0 (versus Fedora 37 which has 7.1.0).  The machine was "ppc64le" and I'm not sure what that is...  but I recall we had problem in the past about regexp differences between x86 and arm, so maybe this is similar...

This was with current release doctest v0.7.0: would be nice to test with current `main` branch but I don't have shell access to the machine :(

```
Symbolic pkg v3.0.0: Python communication link active, SymPy v1.10.1.
Doctest v0.7.0: this is Free Software without warranty, see source.
error: regexp: unrecognized character after (? or (?- at position 6 of expression
error: called from
    doctest_collect>parse_texinfo at line 489 column 8
    doctest_collect>extract_docstring at line 336 column 26
    doctest_collect>collect_targets_class at line 297 column 38
    doctest_collect at line 146 column 11
    doctest_collect at line 126 column 13
    doctest at line 349 column 11
error: Bad exit status from /var/tmp/rpm-tmp.Fi7FuX (%check)
```

Upstream: https://koji.fedoraproject.org/koji/taskinfo?taskID=89344353  (not sure how long those logs are kept).

### PR DESCRIPTION
**Pull request by cbm755**
Related and maybe Fixes #251.

@mmuetzel I think you did this patch to Octave.  Can you give this a quick review for me?

Question: are they recursive?  Like if I put when in a top-level dir do I need to put them in subdirs?

### CODE REVIEW

**Comment by mmuetzel** [comment]
I'm not sure if they are recursive in general. But they are linked to Octave load path logic. That means you don't need (and shouldn't) add them to `private`, `@class` or `+namespace` folders.

This should fix potential errors for the string literals in the sources of doctest itself. But it will probably not affect how characters are read from the source files that it is testing.

File: inst/private/.oct-config [hunk 1]
```diff
@@ -0,0 +1 @@
+encoding=utf-8
```
Comment (mmuetzel) [inline inst/private/.oct-config:1]: This file should not be added in a `private` folder.

File: NEWS [hunk 2]
```diff
@@ -11,7 +11,9 @@ doctest 0.7.0+
 
   * Functions defined directly in the Octave interpreter can now be tested.
 
-  * Test suite fixes for upcoming Octave 6.
+  * Test suite fixes for Octave 6.
+
+  * Source code is in encoded UTF-8, indicated with `.oct-config` files.
```
Comment (mmuetzel) [inline NEWS:16]: I'm not a native speaker. But the grammar in that sentence sounds a bit odd to me.

**Comment by mmuetzel** [comment]
Should a `.oct-config` file be added to the `util` folder, too?

**Comment by cbm755** [comment]
Thanks for the review!

**Pull request merged**